eng/scripts/validate_go_mod.ps1

```diff
@@ -0,0 +1,23 @@
+Param(
+    [string] $serviceDir
+)
+
+Push-Location sdk/$serviceDir
+
+$goModFiles = Get-ChildItem -Path . -Filter go.mod -Recurse
+
+if ($goModFiles.Length -eq 0) {
+    Write-Host "Could not find a go.mod file in the directory"
+    exit 1
+}
+
+foreach ($goMod in $goModFiles) {
+    $patternMatches = Get-Content $goMod.FullName | Select-String -Pattern "replace "
+    if ($patternMatches.Length -ne 0) {
+        $name = $goMod.FullName
+        Write-Host "Found a replace directive in go.mod file at $name"
+        exit 1
```

Comment:
I would log here and set a variable like `$hasError` to true, then exit after the loop. That way you validate all the go.mod files up front (makes it easier for the pipeline tester).